### ISSUE TITLE
No more detection of startup message since 2024

### ISSUE THREAD

**Issue by sarakha63** [SAIC-iSmart-API/saic-python-mqtt-gateway#124]
Hello, i'm using this project (very good by the way) in my home automation in jeedom since 2 months. Everything wazs smooth until the begining of 2024. Since then i can note that the startup message that triggers the refresh interval seems not to be detected anymore (or only once a few times). I think it is probably related to saic side but maybe it's a change on their side or else. But with this issue that makes the usage of the project totaly fail as by default the car is only refreshed once a day (wich is normal). basing the refreshes to 1 minute based on the startup message was a clever way.

Did you notice this issue ?

before it never failed, now it's rare that a startup is detected, i have to manually refresh to let the process continue by itself

regards

**Comment by tosate** [comment]
Hi @sarakha63,

my MQTT gateway instance is still receiving the motor start notifications. Nothing has changed with the start of 2024.
Can you please verify in the MG iSMART app wether notifications are still enabled?
You can find this setting in your profile under `alarm settings > vehicle notifications >motor start notification`
Maybe the names are a bit different since I have translated them from what I see in the app with the German language setting.

**Comment by sarakha63** [comment]
Hi @tosate 

Hello, yes they are still enabled. 
It's strange that nothing as changed as i have two cases where things were systematic before and suddenly became erratic. Sometimes it works sometimes it doesnt moreoften (it doesnt)

So in your case you didnt notice any miss since now ?

If this is the case it's quite strange. This morning the notification was catched on both my wife and my friends instance. I will check when they come back from work. Maybe it was related to a saic server issue  concerning some car Ids or some region .

I will come back if needed

**Comment by tosate** [comment]
We had this issue for a limited amount of time in 2023 once. But back then everyone was affected.
In 2024 I didn't miss a starting notification. At least I didn't notice.

**Comment by jmccrohan** [comment]
I have also noticed this issue. I wonder is it a race condition with the new iSMART 2.0 app?

**Comment by sarakha63** [comment]
@jmccrohan ahhh glad to see somebody else.

Because i still have 6 out of Ten Times thé issue as of today.

I tought about the New app too. but as i use some automation system im pretty sure there was no issue before 2024. And i had the app 2.0.1 before christmas.

But i may think of some changes on server side such as subscription change or else. Hope something will be found because this intégration is important on my side for auto stats based charge

**Comment by jmccrohan** [comment]
@sarakha63 I've looked back at my Vehicle Running and Mileage of the Day sensors, and 3rd Jan is the first time I've observed this issue. I could go restoring backups from December for further details, but I didn't notice this issue until this month.

![image](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/assets/315209/6daf902e-73dd-400d-b242-8c75126bcbdb)
Note the vertical jumps where Vehicle Starts were missed.

**Comment by sarakha63** [comment]
@jmccrohan  I agree with your observation. I also noticed that only since somewhere around 5th of january. It never happened before. and since it happened around 20-30 times.

I'm not sure about the relation with ismart 2.x has i have updated to 2.x somewhere around 20th december

regards

**Comment by nanomad** [comment]
Looking at the new ismart app, with the gateway off, I don't get vehicle start messages there either. The alarm settings are fine so this is probably due to a change on saic side

**Comment by cybersmart-eu** [comment]
that would be no fun then for the Gateway to trigger :-( My whole gsteway based status system is now broken as the gateway does not get any data anymore after a while (due to missing start messages)

**Comment by nanomad** [comment]
> that would be no fun then for the Gateway to trigger :-( My whole Gateway based status system is now broken as the Gateway does not get any data anymore after a while (due to missing start messages) …

I did work today though, I bet it's just SAIC servers doing silly things. As a suggestion, you can use an NFC tag placed somwhere in the car (phone shelf) that triggers an automation on HA side which in turns waits a while then sets the gateway mode to force :)

**Comment by sarakha63** [comment]
Yes Hope SAIC will change.

The strange thing IS that this IS totally random.

Sometimes it triggers sometimes it will miss 5 in a row 

And so on.

**Comment by sarakha63** [comment]
Hello, 

If i go on ismart on the message menu i can see on command tab some 

Privacy Notice Declined followed by a Privacy Notice Declined


I have this on 10/01/2014   15/01/2014   and 17/01/2014  

I dont know if it is linked to the issue lot of us encounter


Fo now i'm still having around 2 out of 3 misses of the start event 

Regards

**Comment by tisoft** [comment]
The new api does not log out the old api and vice versa. As soon as the gateway is converted to use the new api, you will unfortunately be logged out again. Am 22.01.2024 um 12:42 schrieb sarakha63 ***@***.***>:﻿
Hello
another clue or hint
since ismart 2.x  i no more had the token expired please reconnect on my phone even with saic-mqtt working. It only happened if i logged in on my wifes phone and vice versa
before saic-mqtt was acting like a phone and when it connected to saic server it logged me out of the phone (this was expected)
but now it doesnt do that.  It seems the handling of sessions has changed.

**Comment by sarakha63** [comment]
@tisoft 

Do you think this could explain the often miss of the start message évent

Regards

**Comment by nanomad** [comment]
> @nanomad
> 
> Do you think this could explain the often miss of the start message évent
> 
It may be. 

> 
> is your fork using saic-client-ng testable ? I'm a developper i'm willing to help diagnose

It is, I've also published the docker images already so feel free to try. It has almosto feature parity with the current gateway

**Comment by sarakha63** [comment]
> > @nanomad
> > Do you think this could explain the often miss of the start message évent
> 
> It may be.
> 
> > is your fork using saic-client-ng testable ? I'm a developper i'm willing to help diagnose
> 
> It is, I've also published the docker images already so feel free to try. It has almosto feature parity with the current gateway

Ok great. I will install it tomorrow and will make feedbacks.

I actually use the gateway in docker so it will be easy  to Switch.

Thanks

**Comment by sarakha63** [comment]
hello @nanomad  

i cant seem to find the docker image i tried using ghcr.io  but unfortunately i got permission denied

building my self doesnt work as  pypi doesnt have the new saic-ng lib

where did you publish the image you mentioned ?

Regards

**Comment by sarakha63** [comment]
Unfortunately just migrated on the new saic api and still having missing startup events

i dont know what saic did but this is annoying as this disrupts all the logic behind the gateway...

**Comment by nanomad** [comment]
> i dont know what saic did but this is annoying as this disrupts all the logic behind the gateway...

I have no idea either, but messages are not reaching the native app either so there's nothing much we can do. I've been migrating the detection of the car start over to a HA automation. Right now I've settled to use the HA companion app, detect when android auto is running, waiting a bit and then triggering a refresh

**Comment by sarakha63** [comment]
> > i dont know what saic did but this is annoying as this disrupts all the logic behind the gateway...
> 
> I have no idea either, but messages are not reaching the native app either so there's nothing much we can do. I've been migrating the detection of the car start over to a HA automation. Right now I've settled to use the HA companion app, detect when android auto is running, waiting a bit and then triggering a refresh



I did the same for now with automate on Android on my phone and my wifes phone. If it connects to the car bluetooth it sends an info on my home automation system (Jeedom) and then it triggers a force refresh.


By the way the New api implementation IS way faster.

I have the vehicule status and the charging status within 5s of each other . 

Before it was around 35-40 seconds. As each request was long to get the results

**Comment by sarakha63** [comment]
I think this issue can be closes as i think that since the last maintenance in SAic Server (those last days) seemed to have solved the issue, for now it did'nt miss any message, and  before (even with api v2) it was missing at least 2 out of 3.

we can let this issue opened a few days to see and be sure and see the feedbacks of others. but i think it's ok now

**Comment by nanomad** [comment]
I'll close it as it was not something related to the gateway anyway
